src/commands/secrets/download.ts

```diff
@@ -67,6 +53,6 @@ export default class SecretsDownload extends Command {
       }
```

Comment:
This code is the issue.
`fs.writeFile` uses the old callback style and not the `Promise` style of async execution. The `await` here actually does not do anything. Instead the write goes out, but because oclif has no concept of the callback function in the `writeSync` it just goes a head with the test.
Instead you can use this library to get async support.
https://www.npmjs.com/package/fs-extra